main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala

```diff
@@ -133,9 +151,9 @@ object BuildServerProtocol {
     bspBuildTargetDependencySources / aggregate := false,
     bspBuildTargetCompile := Def.inputTaskDyn {
       val s: State = state.value
-      val workspace = bspWorkspace.value
       val targets = spaceDelimited().parsed.map(uri => BuildTargetIdentifier(URI.create(uri)))
-      val filter = ScopeFilter.in(targets.map(workspace))
+      val workspace = bspWorkspace.value.filter(targets)
+      val filter = ScopeFilter.in(workspace.scopes.values.toList)
```

Comment:
Would be nice to log a warning if `bspWorkspace.builds` is not empty so that a warn notification is sent to the build client.

Also in scalaMainClasses, scalaTestClasses and test.

Reply:
Done.